# cpp3d_t2d
Examining overlaps and hypothetical contributions of altered functional capacities in degraded ecosystem soil microbiomes to metabolic anomalies encoded in type 2 diabetes (T2D) gut microbiomes

**Contains code for the article:**

*Degraded ecosystem soil and type 2 diabetes gut microbiomes share altered potential metabolism for sugars, lignin and branched-chain fatty acids: a blind spot for global health?*

by Craig Liddicoat, Bart A. Eijkelkamp, Timothy R. Cavagnaro, Jake M. Robinson, Kiri Joy Wallace, Andrew D. Barnes, Garth Harmsworth, Damien J. Keating, Robert A. Edwards and Martin F. Breed

Preprint server: [BIORXIV/2025/642605](https://doi.org/10.1101/2025.03.11.642605)

**About this repository:** This repository documents code used in the above article, and is not intended for ongoing code development. Sub-folders contain separate workflow documentation and scripts relevant to the case study datasets described below.

**Raw data sources:** comprise type 2 diabetes data from [Forslund et al 2015](https://www.nature.com/articles/nature15766), based on earlier cohort data published by [Karlsson et al 2013](https://www.nature.com/articles/nature12198) from Sweden and [Qin et al 2012](https://www.nature.com/articles/nature12198) from China; together with post-mining ecosystem restoration case study data published by [Sun and Badgley 2019](https://www.sciencedirect.com/science/article/abs/pii/S0038071719301385?via%3Dihub).

**Method overview:** This study uses microbial metagenome functional profiling at the resolution of individual compounds, as outlined in [cpp3d](https://github.com/liddic/cpp3d) and the [manuscript](https://doi.org/10.1101/2025.03.11.642605). [SUPER-FOCUS](https://github.com/metageni/SUPER-FOCUS) functional profile data were generated from the [Swedish](https://github.com/liddic/compound_potential/tree/main/forslund-t2d/ft2d_3_superfocus_fxns) and [Chinese](forslund-t2d-chn/3_fxn_superfocus) cohorts, then subsequently visualised and analysed via **R code**.
